### DISCUSSION TITLE
Briding two network namespaces using StrongSwan

### DISCUSSION THREAD

**Discussion by grahamqs** [strongswan/strongswan#1131] [General]
Hi

Cisco have a nice feaure called VRF aware IPsec, where you can have a VPN in two VRFs. So cipher text is on one VRF and cleartext is on another, and the way that traffic moves between the VRFs is when the tunnel is up, when traffic is encrypted/decrypted it moves between VRFs. This way an attacker would need to bridge an established VPN to move from the outside VRF to the inside VRF.

Looking at SS and Name Space, it looks like SS needs to be in a single namespace and there's no way to 'bridge' namespaces like you can with Cisco VRFs?

many thanks

**Comment by Thermi** [comment]
strongSwan does not process the traffic, the kernel does. The restriction is posed by the kernel.

**Reply by grahamqs** [reply]
Nice one Thermi - thank you.

**Comment by tobiasbrunner** [comment]
Did you look into [XFRM interfaces](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_xfrm_interfaces_on_linux)?

**Reply by grahamqs** [reply]
A friend messaged me about;

https://ifstate.net/examples/xfrm-vrf.html

I will look at this once I start labbing this up.

cheers

**Comment by Thermi** [comment]
I am not so sure this is going to work by simply moving the xfrm interface into the other namespace after configuring it. What does the code say? Of course it might work with policy based routing for the traffic going to and from the other namespace. 

Am 11. Juli 2022 07:24:58 UTC schrieb Tobias Brunner ***@***.***>:
>Did you look into [XFRM interfaces](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_xfrm_interfaces_on_linux)?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/1131#discussioncomment-3119069
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile
